### PR TITLE
[No QA] Create Money Request Header component

### PR DESCRIPTION
**Pull request by yuwenmemon**
@mountiny @luacmartins please review

Right now to get this to show I do a couple of things

#### Web
1. Find an IOU report, since this has the same structure as the type of report we'll be showing here
2. Get the reportID for it
3. Apply this diff:
```diff
diff --git a/src/pages/home/ReportScreen.js b/src/pages/home/ReportScreen.js
index df5ba50e62..83aed35fed 100644
--- a/src/pages/home/ReportScreen.js
+++ b/src/pages/home/ReportScreen.js
@@ -39,6 +39,7 @@ import personalDetailsPropType from '../personalDetailsPropType';
 import getIsReportFullyVisible from '../../libs/getIsReportFullyVisible';
 import EmojiPicker from '../../components/EmojiPicker/EmojiPicker';
 import * as EmojiPickerAction from '../../libs/actions/EmojiPickerAction';
+import MoneyRequestHeader from '../../components/MoneyRequestHeader';
 
 const propTypes = {
     /** Navigation route context info provided by react navigation */
@@ -241,6 +242,16 @@ class ReportScreen extends React.Component {
         // (which is shown, until all the actual views of the ReportScreen have been rendered)
         const shouldAnimate = !shouldFreeze;
 
+        const moneyRequestHeader = this.props.report.type === 'iou' && this.props.report.reportID === '5571340985182399'
+            ? <MoneyRequestHeader report={this.props.report} policies={this.props.policies} personalDetails={this.props.personalDetails} />
+            : (<HeaderView
+                reportID={reportID}
+                onNavigationMenuButtonClicked={() => Navigation.navigate(ROUTES.HOME)}
+                personalDetails={this.props.personalDetails}
+                report={this.props.report}
+                policies={this.props.policies}
+            />);
+
         return (
             <ScreenWrapper
                 style={screenWrapperStyle}
@@ -273,13 +284,7 @@ class ReportScreen extends React.Component {
                                     errors={addWorkspaceRoomOrChatErrors}
                                     shouldShowErrorMessages={false}
                                 >
-                                    <HeaderView
-                                        reportID={reportID}
-                                        onNavigationMenuButtonClicked={() => Navigation.navigate(ROUTES.HOME)}
-                                        personalDetails={this.props.personalDetails}
-                                        report={this.props.report}
-                                        policies={this.props.policies}
-                                    />
+                                    {moneyRequestHeader}
                                 </OfflineWithFeedback>
                                 {Boolean(this.props.accountManagerReportID) && ReportUtils.isConciergeChatReport(this.props.report) && this.state.isBannerVisible && (
                                     <Banner
```
4. Replace `5571340985182399` with the reportID of your IOU
5. Navigate to that report in your browser URL, i.e. http://localhost:8080/r/5571340985182399
6. If you want to test the Expense Report Only Features you can apply this diff
```diff
diff --git a/src/libs/ReportUtils.js b/src/libs/ReportUtils.js
index 7aae22feee..0f03fc99d1 100644
--- a/src/libs/ReportUtils.js
+++ b/src/libs/ReportUtils.js
@@ -105,7 +105,7 @@ function getReportParticipantsTitle(logins) {
  * @returns {Boolean}
  */
 function isExpenseReport(report) {
-    return lodashGet(report, 'type') === CONST.REPORT.TYPE.EXPENSE;
+    return true;
 }
```

#### Mobile
1. In addition to the diff above add the following so that every time you click and IOU you load the MoneyRequestHeader instead:
```DIFF
diff --git a/src/components/ReportActionItem/IOUAction.js b/src/components/ReportActionItem/IOUAction.js
index a7fe3f1710..6835776a87 100644
--- a/src/components/ReportActionItem/IOUAction.js
+++ b/src/components/ReportActionItem/IOUAction.js
@@ -67,7 +67,7 @@ const defaultProps = {
 
 const IOUAction = (props) => {
     const launchDetailsModal = () => {
-        Navigation.navigate(ROUTES.getIouDetailsRoute(props.chatReportID, props.action.originalMessage.IOUReportID));
+        Navigation.navigate(ROUTES.getReportRoute(props.action.originalMessage.IOUReportID));
     };
 
     const shouldShowIOUPreview = (
```
2. To view the Header, click "Pay" on the IOU preview:
Screenshot 2023-05-01 at 3 56 54 PM

### Details
Adds the Money Request Header component into the code so that it can be used in Money Request Flows.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/270587

### Tests
1. Use the instructions above to apply a diff and load an IOU report to display in the header.
2. Load the header
3. Click the three dot menu item makes sure it load the menu in the correct spot
4. On Mobile or any other small screen, make sure that a back button is rendered and that takes you back to the Chat List Menu. On large screens (web and desktop) make sure there is no back button present.

### QA Steps
- No QA - this cannot be as there's no currently logic that loads this component

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

Screenshot 2023-05-01 at 4 14 05 PM

</details>

<details>
<summary>Mobile Web - Chrome</summary>

Screenshot 2023-05-01 at 8 47 52 PM

</details>

<details>
<summary>Mobile Web - Safari</summary>

Screenshot 2023-05-01 at 9 25 54 PM

</details>

<details>
<summary>Desktop</summary>

Screenshot 2023-05-01 at 8 08 42 PM

Note: The Header gap here is a different color. This is because the `ScreenWrapper` component in `ReportScreen` does not have a background color set and we'll most likely need to pass it in as a prop. However, because we're not showing the component yet I'm not sure this is the time to take care of this - since this exists outside of the component. 

</details>

<details>
<summary>iOS</summary>

Screenshot 2023-05-03 at 3 19 14 PM

</details>

<details>
<summary>Android</summary>

Screenshot 2023-05-03 at 3 19 32 PM

</details>

### CODE REVIEW

File: src/components/MoneyRequestHeader.js
Comment (luacmartins) [inline src/components/MoneyRequestHeader.js:57]: We should also show the green checkmark next to the amount if `isSettled = true`

File: src/components/MoneyRequestHeader.js
Comment (luacmartins) [inline src/components/MoneyRequestHeader.js:88]: In the IOU case, we should display the avatar and details of the payee here.

File: src/components/HeaderWithCloseButton.js
Comment (luacmartins) [inline src/components/HeaderWithCloseButton.js:181]: In the IOU case, we should display the avatar and details of the payer here

**Review by luacmartins — COMMENTED** [review]
Looking good. Left a few comments.

File: src/components/HeaderWithCloseButton.js
Comment (yuwenmemon) [reply src/components/HeaderWithCloseButton.js:181]: I believe we do, right? Or maybe can you show me how we're not showing the details of the payer?
https://github.com/Expensify/App/blob/b31c2d136ba3292d0f96c6e7f641539dcf103736/src/libs/ReportUtils.js#L723-L729C6

File: src/components/HeaderWithCloseButton.js
Comment (bondydaa) [inline src/components/HeaderWithCloseButton.js:177]: I know you didn't write this but should this be extracted into a `useCallback` or `useMemo` hook, I think yes but not 100% sure.

File: src/components/MoneyRequestHeader.js
Comment (bondydaa) [inline src/components/MoneyRequestHeader.js:50]: whys this code commented out?

**Review by bondydaa — CHANGES_REQUESTED** [review]
couple of things i noticed, the hook comment is possibly NAB

File: src/components/HeaderWithCloseButton.js
Comment (yuwenmemon) [reply src/components/HeaderWithCloseButton.js:177]: Wouldn't we do that when we refactor this component to use Hooks?

File: src/components/MoneyRequestHeader.js
Comment (yuwenmemon) [reply src/components/MoneyRequestHeader.js:50]: `isSettled` does not exist yet. It will be added as a part of the changes made [here](https://docs.google.com/document/d/17jFcMsAotXvsMhXivoF3u7fyzeyXkKpRGXR88Ud0oWY/edit#bookmark=id.mypo05n9x3ea). This component won't be used anywhere for the moment but it is needed for other components that will link to it. So I figured we'd have this here as an indicator of what this variable would be pulled from later. But happy to adjust however we see fit!

File: src/components/HeaderWithCloseButton.js
Comment (bondydaa) [reply src/components/HeaderWithCloseButton.js:177]: oh doh, i thought this was but must not have realized I moved into a different component when scrolling down the review window 🤦

File: src/components/MoneyRequestHeader.js
Comment (bondydaa) [reply src/components/MoneyRequestHeader.js:50]: oh gotcha, thought maybe it was just leftover from some testing. 

I guess maybe just adding the comment at the end of the line as a todo would be more clear?
```suggestion
    const isSettled = false; // TODO: use ReportUtils.isSettled(props.report.reportID) once that method is added
```

**Review by mountiny — DISMISSED** [review]
This is looking way better to me, great job on this @parasharrajat can you please prioritize the checklist here, I assume we dont have to do tests everywhere and for everythign given this cannot be easily used in app yet. 

@yuwenmemon also lets just update the comment to make ti clear as a TODO in future as @bondydaa mentioned. thanks!

**Comment by JmillsExpensify** [comment]
@parasharrajat I wanted to give you the extra context that this PR is blocking a set of downstream PRs that tie to functionality that we're trying to release in two weeks. It would be awesome if you're able to prioritize this one today so we can get it merged.

File: src/components/HeaderWithCloseButton.js
Comment (luacmartins) [reply src/components/HeaderWithCloseButton.js:185]: Ah yea, my bad!

File: src/components/MoneyRequestHeader.js
Comment (luacmartins) [inline src/components/MoneyRequestHeader.js:82]: We should translate this text

```suggestion
                <Text style={[styles.textLabelSupporting, styles.lh16]}>{props.translate('common.to')}</Text>
```

**Review by luacmartins — CHANGES_REQUESTED** [review]
Changes look good. Left a few comments. It's testing well on web/desktop/mWeb, but I'm having some issues on native, maybe it's related to the data fetching for those reports, although the 2nd avatar seems to in the wrong position.

Web
https://user-images.githubusercontent.com/22219519/235888752-aae78b1c-6c38-4e89-9f39-5fceebea0461.mov

Screenshot 2023-05-03 at 12 04 15 PM

Native
Screenshot_1683109242

File: src/components/AvatarWithDisplayName.js
Comment (grgia) [reply src/components/AvatarWithDisplayName.js:103]: If you remove it it will still hold space for it and not center the title

Comment (yuwenmemon) [reply src/components/AvatarWithDisplayName.js:103]: Screenshot 2023-05-03 at 10 32 52 AM

@grgia just tested and it seems to work fine for me!

**Comment by yuwenmemon** [comment]
Updated, looking into what could be causing the discrepancy between native and web there, can't seem to figure it out (was looking yesterday as well) 🤔

**Comment by yuwenmemon** [comment]
> maybe it's related to the data fetching for those reports, although the 2nd avatar seems to in the wrong position.

I think it's definitely styling... but banging my head against the wall on how to fix it 🤔

**Comment by yuwenmemon** [comment]
Allright I've flexed the fuck out of this bad boy and got the headers looking like normal on Native again!

File: src/components/MoneyRequestHeader.js
Comment (yuwenmemon) [reply src/components/MoneyRequestHeader.js]: I think we have bigger plans for this component so I think we wanted the flexibility. cc @mountiny and @luacmartins who had a bigger hand in the high level design of this.

**Review by grgia — COMMENTED** [review]
@yuwenmemon Looking good! Only comment- should the `<` button be present on all widths? Or only mobile width? 

| Web | Desktop | IOS | Safari | Android | Chrome |
|-----|-----|-----|-----|-----|-----|
|image|image|image|image|image|image|

**Comment by grgia** [comment]
Could you add test steps to QA on workspace vs DM, and also paid versus unpaid (and any other cases where there are visual differences) 

If the `<` button should only be present on certain screen sizes, let's add that as a check too

File: src/components/AvatarWithDisplayName.js
Comment (luacmartins) [reply src/components/AvatarWithDisplayName.js]: @grgia can you confirm if removing the ternary works?

File: src/components/MoneyRequestHeader.js
Comment (Julesssss) [reply src/components/MoneyRequestHeader.js:96]: In the interest of unblocking other issues, maybe we can split that into a separate issue?

File: src/components/AvatarWithDisplayName.js
Comment (grgia) [reply src/components/AvatarWithDisplayName.js]: Removing it does not:  (notice how top header is misaligned).
Also, I _think_ it's preferred to use `condition ? () : null ` rather than `condition && ()` for conditional rendering? But might need to double check that

 image

File: src/components/MoneyRequestHeader.js
Comment (luacmartins) [reply src/components/MoneyRequestHeader.js:96]: I think this is resolved

**Comment by luacmartins** [comment]
> BUG: The tooltip point is not aligned with the icon.
> 
> Screenshot 2023-05-03 at 4 08 03 PM
> 
> 

This seems pretty minor and something we can tackle on a follow up. I won't block on it.

**Comment by luacmartins** [comment]
> @yuwenmemon Looking good! Only comment- should the `<` button be present on all widths? Or only mobile width? 
> 
> 
> 
> | Web | Desktop | IOS | Safari | Android | Chrome |
> 
> |-----|-----|-----|-----|-----|-----|
> 
> |image|image|image|image|image|image|

@grgia the left caret should only be visible on native and mWeb since it takes you back to the LHN and that would be redundant on web/desktop.

File: src/components/AvatarWithDisplayName.js
Comment (luacmartins) [reply src/components/AvatarWithDisplayName.js]: Thanks for confirming. Let's keep the ternary then.

**Review by luacmartins — DISMISSED** [review]
LGTM

**Review by Julesssss — DISMISSED** [review]
It's looking good as far as the tests go. Is the plan to merge it to unblock further changes, rather than holding this up?

**Comment by luacmartins** [comment]
I think so. Just waiting on the checklist

**Review by grgia — DISMISSED** [review]

File: src/components/HeaderWithCloseButton.js
Comment (yuwenmemon) [reply src/components/HeaderWithCloseButton.js:168]: Explained below down here: https://github.com/Expensify/App/pull/18148/files/7d536701547969e16f47104bc55539703e9ad9ac#r1182854883

>  We want the AvatarWithDisplay to be grouped with the back button rather than in the center of the header.

**Comment by yuwenmemon** [comment]
@parasharrajat apologies but what error are you getting?

**Comment by yuwenmemon** [comment]
Okay - @parasharrajat do you want me to do something about it? What are you getting at exactly? I tested and I did see a different JS error which I fixed - but I'm not seeing what you're reporting.

**Comment by JmillsExpensify** [comment]
Can we pull out whatever remaining issues we're seeing, get a checklist in this PR, and then merge? Unfortunately this PR holds up a cascading series of downstream PRs for our upcoming conference. I'd love to get this merged and still keep working on whatever issues we're finding separately.

**Comment by JmillsExpensify** [comment]
Also for what it's worth, we are going to purposefully break IOUs in the coming week.

**Comment by yuwenmemon** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

File: src/components/HeaderWithCloseButton.js
Comment (bondydaa) [inline src/components/HeaderWithCloseButton.js:101]: NAB but if these are only used when `shouldShowAvatarWithDisplay` is `true` then maybe that's a case for having a completely separate component `HeaderWithCloseButtonAndAvavtar` or something like that 🤷.

**Review by bondydaa — APPROVED** [review]
👍 my one comment is def NAB and is maybe something we consider as we keep iterating on this.

**Comment by bondydaa** [comment]
Okay I'm merging this, for the other reviews on here if you have outstanding concerns please spin up new issue(s) for them and we can get them addressed then.

**Comment by hungvu193** [comment]
I was going report the issue but then slack conversation leaded me here.
I think this line broke the header.
https://github.com/Expensify/App/blob/5f649d5045762177c6a6db08265d2d82655bce71/src/components/HeaderWithCloseButton.js#L168
Screenshot 2023-05-05 at 10 39 41

File: src/components/HeaderWithCloseButton.js
Comment (situchan) [reply src/components/HeaderWithCloseButton.js:168]: This causes regression: All RHN header styles are broken

rhn header

**Comment by yuwenmemon** [comment]
Ah, my b. Will look at this.

**Comment by luacmartins** [comment]
@yuwenmemon feel free to assign me for review

**Comment by JmillsExpensify** [comment]
Yep, our bad.

**Comment by Santhosh-Sellavel** [comment]
We missed a case here, which led to this [issue](https://github.com/Expensify/App/issues/19340)

**Pull request merged**
